### ISSUE TITLE
Right Click Context Menu Between Two Windows Machines On Wifi Causes Consist Freezing of 10 Seconds

### ISSUE THREAD

**Issue by othmanalikhan** [debauchee/barrier#1689]
### What happened?

Topology
----------
Laptop (client) <---2.4 GHz---> AP <---> Router <---> AP <--5 GHz---> PC (host)

Steps to Reproduce
----------------------
1. On the client machine, right-click any file on the desktop.
2. Mouse becomes immovable and jams for 10 seconds on client machine.

Investigation
--------------
1. Wireshark capture and Barrier debug logs indicates connection between both machines is lost and re-established during those 10 seconds of down time.
2. While the mouse is jammed for 10 seconds, Barrier packets are actually still sent and received between both machines.
3. Disabling and enabling SSL doesn't change the behaviour.
4. Using certain shortcuts like Windows Key + E also triggers this behaviour.
5. Switching to a wired connection between the two machines stops the issue

Explanation
-------------
1. It seems that the rate of packets being transmitted between the two machines is rather high, and Wifi supposedly has issues when the rate spikes (when you right-click open context menu on a file). Perhaps using more antennas or full 5 GHz might fix the issue, otherwise sticking with a wired Ethernet connection is the solution for now.
2. There is a possibility that the issue itself could be the Barrier application can't cope with the 'unstable' network connection on Wifi. An easy way to verify this would be to use an app that mimics lag (e.g. Clumsy on Github) on a wired connection, and see if the behaviour is reproducible. 

### Version

v2.4.0

### If applicable, where did you install Barrier from?

Downloaded `2.4.0-release-3e0d758b` as pre-built binary from Github.

### What OSes are you seeing the problem on? (Check all that apply)

Windows

### What OS versions are you using?

PC
---
- OS: Microsoft Windows 10 Pro
- Version:	10.0.19044 Build 19044

Laptop
--------
- OS: Microsoft Windows 10 Home
- Version: 10.0.19043 Build 19043

**Comment by othmanalikhan** [comment]
This issue has a workaround solution which is to stick with a wired connection instead of Wifi, but I thought I would post this mouse jamming behaviour for anyone who is suffering in silence.

**Comment by Jean757** [comment]
@OthmanEmpire Are you sure that the issue is related to Barrier and not your entire pc, because some week ago i had the exact same issue (wifi freezing when right clicking icon on the desktop) and i solved it by uninstalling all devices named " Microsoft Wifi Direct Adapter " (visible and hidden too) in my devices manager.
Barrier was not installed at this time so i'm sure it was not related to

**Comment by othmanalikhan** [comment]
> @OthmanEmpire Are you sure that the issue is related to Barrier and not your entire pc, because some week ago i had the exact same issue (wifi freezing when right clicking icon on the desktop) and i solved it by uninstalling all devices named " Microsoft Wifi Direct Adapter " (visible and hidden too) in my devices manager. Barrier was not installed at this time so i'm sure it was not related to

Interesting. What does Wifi freezing mean in your scenario? Does the entire machine freeze or the mouse unusable? I think that might be different to my case since I can move my mouse on the remote machine if I plug in another mouse (but the Barrier 'mouse' freezes so I can't move it remotely).

Barrier works seamlessly between my two machines when they are both connected on Ethernet, so I would say that rules out anything funny going on my machines except for network/NIC related issues. Perhaps later I should try plugging in an external Wifi USB on my laptop in particular and test whether the issue persists on Wifi.

**Comment by othmanalikhan** [comment]
So I've just retested producing this bug with another Wifi NIC (just a Wifi USB NIC) and tried both on 2.4 GHz and 5 GHz and I'm unable to reproduce the issue. I've tried also reproducing it with the original Wifi card on my laptop but it just magically disappeared (even though I've been having the issue for about 1 year now). I'm not too sure what is going on but I will just chalk up this issue just as resolved.

**Comment by othmanalikhan** [comment]
So the issue started manifested again after few months, and I can confirm the issue is stems from Wifi.

Steps to reproduce the issue:

1. Move the mouse to the remote machine (in this case my laptop).
2. Just launch file explorer and wait 3 seconds.
3. Mouse freezes on all machines and is released after 10 seconds.
4. Barrier logs indicate that the 'server is dead' and WireShark confirms TCP re-transmissions.

The issue in particular is not related to the packet drops per-se nor bandwidth of the network connection, but rather, how 2.4 GHz handles a burst of packets and how Barrier handles failure of transmission of these packets. 

In short, just switching to 5 GHz (more antennas and other Wifi technologies) solves the issue. I'm leave this message here though for other people facing similar issues.
